src/routes/apps/chat-app/message.routes.js

```diff
@@ -24,4 +25,19 @@ router
     sendMessage
   );
 
+//attachment id is optional for multiple attachments in a single message
+
+router.route("/:chatId/:messageId/:attachmentId?").delete(
+  mongoIdPathVariableValidator("chatId"),
+  mongoIdPathVariableValidator("messageId"),
+  (req, res, next) => {
+    if (!req.params.attachmentId) {
+      return next();
+    }
+    mongoIdPathVariableValidator("attachmentId")[0](req, res, next);
```

Comment:
No need to add attachments level deletion, attachments are part of message so if anyone want to delete a message then we will delete entire message including attachments no need to add granular deletion for attachments